src/main.cpp

```diff
@@ -857,6 +1026,57 @@ int main(int argc, char *argv[])
     wl_registry_add_listener(registry, &registry_listener, NULL);
     sync_wayland();
 
+    if (params.codec.find("vaapi") != std::string::npos)
```

Comment:
There are many cases here, some of which may not be apparent to the user. I think it would be nice to print messages not just in the error cases, but basically every time we make a decision: therefore the user can see what is going on internally (e.g. when we use dmabuf, whether we use the same device as compositor, etc).